src/components/Wrappers/withClaimReturned.js

```diff
@@ -15,6 +66,180 @@ const withClaimReturned = WrappedComponent => class withClaimReturnedComponent e
+        }
+      } = this.props;
+      const lostStatus = 'Lost item fee';
+      const processingStatus = 'Lost item processing fee';
+      const pathParts = [
+        'accounts?query=',
+        `barcode=="${itemBarcode}"`,
+        ` and userId=="${userId}"`,
+        ` and (feeFineType=="${lostStatus}"`,
+        ` or feeFineType=="${processingStatus}")`
+      ];
+      const path = pathParts.reduce((acc, val) => acc + val, '');
+      return GET({ path });
+    };
+
+    const setPaymentStatus = record => {
+      const updatedRec = cloneDeep(record);
+      updatedRec.paymentStatus.name = 'Suspended claim returned';
+      return updatedRec;
+    };
+
+    const persistAccountRecord = record => {
+      const {
+        mutator: {
+          activeAccount: {
+            update,
+          },
+          accounts: {
+            PUT,
+          }
+        }
+      } = this.props;
+      update({ id: record.id });
+      return PUT(record);
+    };
+
+    const getAccountActions = account => {
+      const {
+        mutator: {
+          feefineactions: {
+            GET,
+          }
+        }
+      } = this.props;
+      const path = `feefineactions?query=(accountId==${account.id})&orderBy=dateAction&order=desc`;
+      return GET({ path });
+    };
+
+    const filterTransferredActions = actions => actions
+      .filter(
+        record => record.typeAction && record.typeAction.startsWith('Transferred')
+      );
+
+    const getLastBalance = actions => (
+      actions.length > 0
+        ? orderBy(actions, ['dateAction'], ['desc'])[0].balance
+        : 0.0
+    );
+
+    const createRefundActionTemplate = (account, transferredActions, type) => {
+      const {
+        okapi: {
+          currentUser: {
+            id: currentUserId,
+            curServicePoint: {
+              id: servicePointId
+            }
+          },
+        },
+      } = this.props;
+      const orderedActions = orderBy(transferredActions, ['dateAction'], ['desc']);
+      const now = moment().format();
+      const sign = type.startsWith('Credited') ? -1 : 1;
+      const amount = sign * transferredActions.reduce((acc, record) => acc + record.amountAction, 0.0);
+      const transactionVerb = type.startsWith('Credited')
+        ? 'Refund'
+        : 'Refunded';
+      return {
+        dateAction: now,
+        typeAction: type,
+        comments: '',
+        notify: false,
+        amountAction: amount,
+        balance: 0,
+        transactionInformation: `${transactionVerb} to ${orderedActions[0].paymentMethod}`,
+        source: orderedActions[0].source,
+        paymentMethod: '',
+        accountId: account.id,
+        userId: currentUserId,
+        createdAt: servicePointId,
+      };
+    };
+
+    const createRefunds = (account, actions) => {
+      return actions.length > 0
+        ? [
+          createRefundActionTemplate(account, actions, 'Credited fully'),
+          createRefundActionTemplate(account, actions, 'Refunded fully')
```

Comment:
Can refund-actions be listed in `src/constants.js`?

Reply:
Yes, these strings were moved to constants.js